[WIP] Move detailed information to wiki pages

ritiek opened the pull request:
I've moved some of the detailed information to GitHub wiki pages.

@linusg @vn-ki Feel free to add any other information you guys find necessary to wiki pages. Also let me know if there is anything else we could remove from README. :)

ritiek commented:
> I think we should remove the usage (--help) dump completely from the README, that's something the user can do on their own. Also the most important flags are explained in the README anyway.

I am not sure about this one. I personally like projects who display the help usage in README to give people a rough idea of what all the tool is capable of.

> Maybe shorten the list of features and display it in a way so it takes up less space?

Yep, gotta look in this too.

> I wouldn't say "Specify the Target Directory" should be in "Advanced options" but maybe stay in the README...

This seems a better idea!

ritiek commented:
What about we move sections on downloading by track name, spotify link, playlist, etc. to wiki and keep help usage in README?

linusg commented:
FAQ + linking from the README sounds nice! Keeping the help output and moving all the detailed explanations to the Wiki is fine as long as the help output will not grow and grow and grow... 😉
When doing so I would say just having a "getting started" section with e.g. instructions on how to download a single song by name and a playlist by URL (features I personally use the most) should be sufficient.

vn-ki commented:
> What about we move sections on downloading by track name, spotify link, playlist, etc. to wiki and keep help usage in README?

I think these are better off being in the readme. Most people don't open wiki unless they really need to.

> Also not sure about this but maybe we could have an FAQ section in wiki (for stuff like how to specify the target directory, etc) and link to it from README.

I'm all for an FAQ section.

I haven't taken a look at the wiki yet. I'm a bit busy right now.

ritiek commented:
> I think these are better off being in the readme. Most people don't open wiki unless they really need to.

I think we could specify the most basic `-s` option in README and then provide a link to using other options (`-f`, `-p`, etc) in wiki.

ritiek commented:
(Also I want to thank you both for being so responsive :)

linusg commented:
> I am not sure about this one. I personally like projects who display the help usage in README to give people a rough idea of what all the tool is capable of.

We should probably reconsider this: The output of `python3 spotdl.py --help` is currently almost 70 lines, which indeed only giving a "rough" overview to first-time users.

I certainly understand what you mean, a raw `-h`/`--help` dump can be better than a ton of text, but in our case it seems overwhelming.

Even the first few lines. As a beginner, but avid Linux user/programmer I would love somthing like:

```
(-s SONG | -l LIST | -p PLAYLIST | -b ALBUM | -u USERNAME | -V)
[-f FOLDER] [--overwrite {force,prompt,skip}] [-o OUTPUT_EXT][-c CONFIG]
```

This is totally self explaining. The whole thing probably not:

```
 (-s SONG | -l LIST | -p PLAYLIST | -b ALBUM | -u USERNAME | -V)
[-m] [-nm] [-a] [-f FOLDER] [--overwrite {force,prompt,skip}]
[-i {.webm,.m4a}] [-o OUTPUT_EXT] [-ff FILE_FORMAT]
[-sf SEARCH_FORMAT] [-dm] [-d] [-mo] [-ns]
[-ll {INFO,WARNING,ERROR,DEBUG}] [-c CONFIG]
```

Because "what the hell is `-m`, `-nm`, `-i`, `-dm`, `-d`, `-mo` and `-ns`"? (I know, it's explained later.)

So does the Python `argparse` module offer something like a short help (`-h`/`--help`) and an extended/full help (sth. like `--help-complete`)? I may be wrong, but I think some GNU tools offer this...

ritiek commented:
Okay, I moved the help usage to wiki and added an FAQ section. The README looks super cool now. We could merge this right away unless you guys have any other suggestions!

ritiek commented:
@vn-ki Good point. I added some basic information in 5ee65826bd89f5cbce0afe396285f13c4248ce2c.

ritiek commented:
Okay then, there we go!
